components/movementsensor/gpsutils/vrs.go

```diff
@@ -23,11 +24,13 @@ func ConnectToVirtualBase(ntripInfo *NtripInfo,
 	credentialsBase64 := base64.StdEncoding.EncodeToString([]byte(credentials))
 
 	// Process the server URL
-	serverAddr := ntripInfo.URL
-	serverAddr = strings.TrimPrefix(serverAddr, "http://")
-	serverAddr = strings.TrimPrefix(serverAddr, "https://")
+	serverAddr, err := url.Parse(ntripInfo.URL)
+	if err != nil {
+		logger.Error(err)
+		return nil
+	}
 
-	conn, err := net.Dial("tcp", serverAddr)
+	conn, err := net.Dial("tcp", serverAddr.Host)
 	if err != nil {
 		return nil
```

Comment:
This line was (I believe) the cause of the initial panic. The error just "disappears" and a nil connection is returned. Then things panic when the calling code tries to use the returned value, as it has no way of knowing there was an error.

Reply:
Sorry if I wasn't clear. I'm suggesting these function itself needs to have its signature changed, and actually returns an error, so when problems occur they are actually handled (and passed up), instead of panicking.

Reply:
ah okay I see